util/topgen.py

```diff
@@ -1339,40 +1339,16 @@ def render_template(template_path: str, rendered_path: Path,
             (SRCTREE_TOP / "hw/top_{}/".format(topname)).resolve()
         ]
         for idx, path in enumerate(out_paths):
-            # "clang-format" -> "sw/autogen/.clang-format"
-            cformat_tplpath = TOPGEN_TEMPLATE_PATH / "clang-format"
-            cformat_dir = path / "sw/autogen"
-            cformat_dir.mkdir(parents=True, exist_ok=True)
-            cformat_path = cformat_dir / ".clang-format"
-            cformat_path.write_text(cformat_tplpath.read_text())
-
-            # Save the header macro prefix into `rs_helper`
-            rel_header_dir = cformat_dir.relative_to(root_paths[idx])
-            rs_helper.header_macro_prefix = (
-                "OPENTITAN_" + str(rel_header_dir).replace("/", "_").upper())
-
-            # "top_{topname}.rs.tpl" -> "sw/autogen/top_{topname}.rs"
-            cheader_path = cformat_dir / f"top_{topname}.rs"
-            render_template(TOPGEN_TEMPLATE_PATH / "toplevel.rs.tpl",
-                            cheader_path,
-                            helper=rs_helper)
-
-            # TODO Rust don't have header this path
-            # should be used to generate rust modules.
-            # Save the relative header path into `rs_helper`
-            rel_header_path = cheader_path.relative_to(root_paths[idx])
-            rs_helper.header_path = str(rel_header_path)
-
-            # "toplevel_memory.ld.tpl" -> "sw/autogen/top_{topname}_memory.ld"
-            render_template(TOPGEN_TEMPLATE_PATH / "toplevel_memory.ld.tpl",
-                            cformat_dir / f"top_{topname}_memory.ld")
-
-            # TODO megre to one memmory file
-            # "toplevel_memory.rs.tpl" -> "sw/autogen/top_{topname}_memory.rs.h"
-            memory_cheader_path = cformat_dir / f"top_{topname}_memory.rs"
-            render_template(TOPGEN_TEMPLATE_PATH / "toplevel_memory.rs.tpl",
-                            memory_cheader_path,
-                            helper=rs_helper)
+            rsformat_dir = path / "sw/autogen/chip"
+            rsformat_dir.mkdir(parents=True, exist_ok=True)
+
+            rust_files = [("toplevel_mod.rs.tpl", "mod.rs"),
+                          ("toplevel.rs.tpl", f"top_{topname}.rs"),
+                          ("toplevel_memory.rs.tpl", f"top_{topname}_memory.rs")]
+            for (template, source) in rust_files:
+                render_template(TOPGEN_TEMPLATE_PATH / template,
+                                rsformat_dir / source,
+                                helper=rs_helper)
```

Comment:
@hcindyl Thera are changes similar to your PR: https://github.com/lowRISC/opentitan/pull/17576

Reply:
In that case, I will drop mine.